### PR TITLE
[Android] Add functions of autocapture to run in android Activities

### PR DESCRIPTION
**Pull request by park-sang-hyun**
[Edit] nnstreamer-jni.h : Add data structure to contain conditions for autocapture.
[Edit] nnstreamer-jni.c : Add 'NativeDeleteLineAndLabel', 'NativeInsertLineAndLabel', 'NativeGetCondition', 'NativeGetAutoCapture' functions based on jni.
[Edit] nnstreamer-ex.cpp : Implement 'NativeDeleteLineAndLabel', 'NativeInsertLineAndLabel', 'NativeGetCondition', 'NativeGetAutoCapture' functions.

When objects that meet the conditions are visible on the screen, can take picture automatically.
When in autocapture mode, can erase the label's name and line so that only objects can appear in the picture taken.

Signed-off-by: sang.hyun.park <park03851@naver.com>

### CODE REVIEW

**Comment by macaron10** [comment]
LGTM!

File: android/example_app/nnstreamer-ssd-autocapture/jni/nnstreamer-ex.cpp [hunk 1]
```diff
@@ -392,6 +409,28 @@ ssd_get_detected_objects(ssd_object_s *objects)
...
 }
 
+/**
+ * @brief Compare detected objects with conditions.
+ */
+static bool
+ssd_matching_detect_and_condition(){
+  bool check = true;
+
+  if(setting_object_list.empty() || detected_object_list.empty()) return false;
```
Comment (zhoonit) [inline android/example_app/nnstreamer-ssd-autocapture/jni/nnstreamer-ex.cpp:419]: Have you run clang-format? indentation seems not coherent

File: android/example_app/nnstreamer-ssd-autocapture/jni/nnstreamer-ex.cpp [hunk 2]
```diff
@@ -424,19 +463,40 @@ ssd_draw_object(cairo_t *cr, ssd_object_s *objects, const gint size)
...
       nns_logd("Failed to get label (class id %d)", objects[i].class_id);
     }
   }
+
+  /* send true or false values by comparing detected objects with conditions */
+  if(ssd_matching_detect_and_condition()){
+    is_auto_capture = true;
+  }else{
+    is_auto_capture = false;
+  }
```
Comment (zhoonit) [inline android/example_app/nnstreamer-ssd-autocapture/jni/nnstreamer-ex.cpp:497]:
```suggestion
  is_auto_capture = ssd_matching_detect_and_condition();
```

File: android/example_app/nnstreamer-ssd-autocapture/jni/nnstreamer-ex.cpp [hunk 3]
```diff
@@ -73,10 +75,25 @@ typedef struct
...
+};
+
 static gint launch_option = 0;
 static nns_ex_model_info_s nns_ex_model_info;
 static GMutex res_mutex;
 static std::vector<ssd_object_s> detected_object;
+static std::map<gchararray, gint, cmp_str> detected_object_list;
+static std::map<gchararray, gint, cmp_str> setting_object_list;
+static gboolean is_line_title;
+static gboolean is_auto_capture;
```
Comment (zhoonit) [inline android/example_app/nnstreamer-ssd-autocapture/jni/nnstreamer-ex.cpp:96]: Please add comment on what those variable do

`static gboolean is_auto_capture; /**< ... */

File: android/example_app/nnstreamer-ssd-autocapture/jni/nnstreamer-jni.c [hunk 4]
```diff
@@ -713,6 +714,69 @@ gst_native_get_description(JNIEnv *env, jobject thiz, jint id, jint option)
...
+static void
+gst_native_get_condition_object(JNIEnv *env, jobject _obj, jobjectArray _objects)
+{
+  jsize len = (*env)->GetArrayLength(env, _objects);
+  jobject conditionsObj;
+  jclass clazz;
+  jfieldID fid;
+  jstring jstr;
+
+  SettingData * datas = (SettingData *)malloc(sizeof(SettingData) * len);
```
Comment (zhoonit) [inline android/example_app/nnstreamer-ssd-autocapture/jni/nnstreamer-jni.c:747]: Q. Where does free this?

File: android/example_app/nnstreamer-ssd-autocapture/jni/nnstreamer-ex.cpp [hunk 5]
```diff
@@ -424,19 +463,40 @@ ssd_draw_object(cairo_t *cr, ssd_object_s *objects, const gint size)
...
   }
+
+  /* send true or false values by comparing detected objects with conditions */
+  if(ssd_matching_detect_and_condition()){
+    is_auto_capture = true;
+  }else{
+    is_auto_capture = false;
+  }
+
+  detected_object_list.clear();
```
Comment (zhoonit) [inline android/example_app/nnstreamer-ssd-autocapture/jni/nnstreamer-ex.cpp:499]: It seems that `detected_object_list` doesn't need to be a global variable.

File: android/example_app/nnstreamer-ssd-autocapture/jni/nnstreamer-jni.c [hunk 6]
```diff
@@ -713,6 +714,69 @@ gst_native_get_description(JNIEnv *env, jobject thiz, jint id, jint option)
...
+    fid = (*env)->GetFieldID(env, clazz, "count", "I");
+    int count = (*env)->GetIntField(env, conditionsObj, fid);
+    strcpy(datas[i].name, pName);
+    datas[i].count = count;
+
+    nns_logd("Conditions Data --> %s -- %d \n", pName, count);
+    (*env)->ReleaseStringUTFChars(env, jstr, pName);
+  }
+
+  nns_logd("Error??");
```
Comment (zhoonit) [inline android/example_app/nnstreamer-ssd-autocapture/jni/nnstreamer-jni.c:765]: Unintended log?

**Comment by park-sang-hyun** [comment]
Thank you for your detailed coaching on the parts I missed! 
I'll fix what you said!

File: android/example_app/nnstreamer-ssd-autocapture/jni/nnstreamer-ex.cpp (hunk 5 above)
Comment (park-sang-hyun) [reply android/example_app/nnstreamer-ssd-autocapture/jni/nnstreamer-ex.cpp:506]: 'detected_object_list' was also used in 'ssd_matching_detect_and_condition' and declared global, should it be modified by handing over parameters...?

Comment (zhoonit) [reply android/example_app/nnstreamer-ssd-autocapture/jni/nnstreamer-ex.cpp:506]: I'd recommend it to if possible.

**Comment by zhoonit** [comment]
When you make a revision, we amend the existing commit and force push to the branch.

File: android/example_app/nnstreamer-ssd-autocapture/jni/nnstreamer-ex.cpp (hunk 5 above)
Comment (park-sang-hyun) [reply android/example_app/nnstreamer-ssd-autocapture/jni/nnstreamer-ex.cpp:506]: Thank you for your advice.

**Comment by park-sang-hyun** [comment]
I tried to restore it, but the 'native' commit was stuck in the middle, so when the squash was implemented, the commit had to be merged together.
Should I write a full request again...?

**Comment by zhoonit** [comment]
> I tried to restore it, but the 'native' commit was stuck in the middle, so when the squash was implemented, the commit had to be merged together.
> Should I write a full request again...?

No you just need to rewrite the history. Could you DM me at around 5?

**Comment by park-sang-hyun** [comment]
I'll send you DM around 5. Thank you!

**Comment by macaron10** [comment]
LGTM!

**Comment by Soyeong-k-1230** [comment]
LGTM

**Comment by Joojihwan** [comment]
LGTM

**Comment by JeongYeonUk** [comment]
LGTM!

**Review by zhoonit — APPROVED** [review]
LGTM :)

**Pull request merged**